### PR TITLE
Possible fix for document upload corrupted file bug

### PR DESCRIPTION
**Pull request by amobarak**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

* Hardware:
* O/S and version:
* JDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/assets/javascripts/documentUpload.js [hunk 1]
```diff
@@ -165,7 +165,20 @@
...
+            success: function() {
+              self.removeFileFromList($file);
+              for(var i = 0; i < dropzone.files.length; i++){
+                if(dropzone.files[i].status === 'success'){
+                  dropzone.files.splice(i, 1);
+                  return;
+                }
+              }
+            }
+          });
```
Comment (hosainnet) [inline app/assets/javascripts/documentUpload.js]: Any chance we can test this?

File: app/assets/javascripts/documentUpload.js [hunk 2]
```diff
@@ -165,7 +165,20 @@
         this.$fileList.on('click', '.hide-file', function(e) {
           e.preventDefault();
           var $file = $(this).parents('.file');
-          self.removeFileFromList($file);
+          var fileUrl = $(this).data('fileurl');
+          $.ajax({
+            url: window.location + '?js=true&fileUrl=' + encodeURI(fileUrl) + '&_csrf=' + csrfToken,
```
Comment (alexwakeman) [inline app/assets/javascripts/documentUpload.js]: I feel like we should a) set this URL up somewhere else to make it more readable. b) Use JavaScript template literals:
```
const endPoint = `${window.location}?js=true&fileUrl=${encodeURI(fileUrl)}&_csrf=${csrfToken}`;
```

File: app/assets/javascripts/documentUpload.js [hunk 3]
```diff
@@ -165,7 +165,20 @@
...
+            type: 'DELETE',
+            success: function() {
+              self.removeFileFromList($file);
+              for(var i = 0; i < dropzone.files.length; i++){
```
Comment (alexwakeman) [inline app/assets/javascripts/documentUpload.js]: Spacing around `for(` should be `for (` (with space). Cheers!

Comment (alexwakeman) [reply app/assets/javascripts/documentUpload.js]: Same for `if` statements

File: app/assets/javascripts/documentUpload.js [hunk 4]
```diff
@@ -165,11 +166,25 @@
...
           var $file = $(this).parents('.file');
-          self.removeFileFromList($file);
+          var fileUrl = $(this).data('fileurl');
+          const endPoint = `${window.location}?js=true&fileUrl=${encodeURI(fileUrl)}&_csrf=${csrfToken}`;
+          $.ajax({
+            url: endPoint,
+            type: 'DELETE',
+            success: function() {
+              self.removeFileFromList($file);
+              for (var i = 0; i < dropzone.files.length; i++){
```
Comment (alexwakeman) [inline app/assets/javascripts/documentUpload.js]: Could just use 
```
dropzone.files.forEach(file => {
...
});
```
Just a suggestion

File: app/assets/javascripts/documentUpload.js [hunk 5]
```diff
@@ -165,11 +166,25 @@
...
+                  return;
+                }
+              }
+            }
+          });
         });
 
         $(document).on('keydown', 'a.faux-link, .dz-clickable', function(e) {
-          if([13, 32].includes(e.keyCode)) { // pressed RETURN or SPACE
+          if ([13, 32].includes(e.keyCode)) { // pressed RETURN or SPACE
```
Comment (alexwakeman) [inline app/assets/javascripts/documentUpload.js]: I realise you didn't write this but we should avoid "magic numbers" and declare them formally somewhere appropriate

File: app/assets/javascripts/documentUpload.js (hunk 4 above)
Comment (alexwakeman) [reply app/assets/javascripts/documentUpload.js]: Actually checking this again - I guess a for loop is ok as it returns early

File: test/end-to-end/pages/marriage/upload.js [hunk 6]
```diff
@@ -56,5 +68,6 @@ module.exports = {
   uploadMarriageCertificateFile,
   deleteAMarriageCertificateFile,
   testUploadResponse,
-  withoutUploadFile
+  withoutUploadFile,
+  uploadCorruptedFileAndRemovingSuccessfully
```
Comment (hosainnet) [inline test/end-to-end/pages/marriage/upload.js]: Where is `uploadCorruptedFileAndRemovingSuccessfully` used?

File: app/assets/javascripts/documentUpload.js [hunk 7]
```diff
@@ -148,13 +148,14 @@
           e.preventDefault();
           var $file = $(this).parents('.file');
           var fileUrl = $(this).data('fileurl');
+          const endPoint = `${window.location}?js=true&fileUrl=${encodeURI(fileUrl)}&_csrf=${csrfToken}`;
```
Comment (alexwakeman) [inline app/assets/javascripts/documentUpload.js]: As this is running on the client - we should avoid using `const` as it's ES6 and not supported by older browsers...

File: test/end-to-end/pages/marriage/upload.js [hunk 8]
```diff
@@ -45,6 +45,18 @@ function deleteAMarriageCertificateFile(isDragAndDropSupported) {
...
 
+function uploadCorruptedFileAndRemovingSuccessfully(isDragAndDropSupported) {
+  const I = this;
+
+  I.say('Drag and Drop supported: ' + isDragAndDropSupported);
+  I.seeCurrentUrlEquals('/petitioner-respondent/marriage-certificate-upload');
+  upload.call(I, '/assets/corruptedImage.bmp', isDragAndDropSupported);
+  I.waitForVisible('.file', 30);
+  I.waitForText('Hide', 30);
+  I.see('No files uploaded');
```
Comment (Dizrupto) [inline test/end-to-end/pages/marriage/upload.js]: I think there should be I.click('Hide)' before that line. @hosainnet what do you think?

File: app/assets/javascripts/documentUpload.js [hunk 9]
```diff
@@ -165,11 +166,25 @@
...
+                  return;
+                }
+              }
+            }
+          });
         });
 
         $(document).on('keydown', 'a.faux-link, .dz-clickable', function(e) {
-          if([13, 32].includes(e.keyCode)) { // pressed RETURN or SPACE
+          if ([13, 32].includes(e.keyCode)) { // pressed RETURN or SPACE
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]:
```
const RETURN = 13;
const SPACE = 32;

if ([RETURN, SPACE].includes(e.keyCode))
```

or:

```
if (spaceOrReturnPressed(e.keyCode)) {```

File: app/assets/javascripts/documentUpload.js [hunk 10]
```diff
@@ -148,13 +148,14 @@
...
+          var endPoint = `${window.location}?js=true&fileUrl=${encodeURI(fileUrl)}&_csrf=${csrfToken}`;
           $.ajax({
-            url: window.location + '?js=true&fileUrl=' + encodeURI(fileUrl) + '&_csrf=' + csrfToken,
+            url: endPoint,
             type: 'DELETE',
             success: function() {
               self.removeFileFromList($file);
-              for(var i = 0; i < dropzone.files.length; i++){
-                if(dropzone.files[i].status === 'success'){
+              for (var i = 0; i < dropzone.files.length; i++){
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: white space before `{`

File: app/assets/javascripts/documentUpload.js [hunk 11]
```diff
@@ -122,7 +122,7 @@
         self.$zone.dropzone(dzOptions);
 
         // fix a11y tests
-        if($('.dz-hidden-input').length){
+        if ($('.dz-hidden-input').length){
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: white space before `{`

File: app/assets/javascripts/documentUpload.js [hunk 12]
```diff
@@ -40,48 +40,48 @@
...
             this.on('error', function(file, errorMessage){
               var errorMessageText = options.errors.errorUnknown;
 
-              if(errorMessage && errorMessage.code && options.errors[errorMessage.code]){
+              if (errorMessage && errorMessage.code && options.errors[errorMessage.code]){
                 errorMessageText = options.errors[errorMessage.code];
               }
 
-              if(file.element){
+              if (file.element){
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: white space before `{`

File: app/assets/javascripts/documentUpload.js [hunk 13]
```diff
@@ -40,48 +40,48 @@
...
                 file.element.addClass('error');
                 file.element.find('span.form-hint').remove();
                 file.element.find('td:first')
                   .addClass('form-group-error')
                   .append('<span class="error-message">' + options.errors.errorUnknown + '</span>');
               }
-              if(!this.getQueuedFiles().length){
+              if (!this.getQueuedFiles().length){
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: white space before `{`

File: app/assets/javascripts/documentUpload.js [hunk 14]
```diff
@@ -40,48 +40,48 @@
...
             });
             this.on('success', function(file, resp){
-              if(file.element && resp && resp.length){
+              if (file.element && resp && resp.length){
                 $(file.element).find('.remove-file').data('fileurl', resp[0].fileUrl);
               }
             });
             this.on('canceled', function(file){
-              if(file.element){
+              if (file.element){
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: white space before `{`

File: app/assets/javascripts/documentUpload.js [hunk 15]
```diff
@@ -40,48 +40,48 @@
...
             this.on('uploadprogress', function(file, progress){
-              if(file.element){
+              if (file.element){
                 file.element.find('span.progress').attr('style', 'width:'+progress+'%;');
                 file.element.find('span.form-hint').html(' - ' + options.uploadingHint);
               }
             });
             this.on('complete', function(file){
-              if(file.element){
+              if (file.element){
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: white space before `{`

File: app/assets/javascripts/documentUpload.js [hunk 16]
```diff
@@ -40,48 +40,48 @@
               $('.dz-preview').remove();
             });
             this.on('uploadprogress', function(file, progress){
-              if(file.element){
+              if (file.element){
```
Comment (yarpo) [inline app/assets/javascripts/documentUpload.js]: white space before `{`

**Comment by dtaylor7** [comment]
closing due to age

**Pull request closed without merging**